elixir/apps/web/test/web/live/settings/dns/index_test.exs

```diff
@@ -0,0 +1,156 @@
+defmodule Web.Live.Settings.DNS.IndexTest do
+  use Web.ConnCase, async: true
+
+  setup do
+    Domain.Config.put_system_env_override(:outbound_email_adapter, Swoosh.Adapters.Postmark)
+
+    account = Fixtures.Accounts.create_account()
+    identity = Fixtures.Auth.create_identity(account: account, actor: [type: :account_admin_user])
+
+    %{
+      account: account,
+      identity: identity
+    }
+  end
+
+  test "redirects to sign in page for unauthorized user", %{account: account, conn: conn} do
+    assert live(conn, ~p"/#{account}/settings/dns") ==
+             {:error,
+              {:redirect,
+               %{
+                 to: ~p"/#{account}",
+                 flash: %{"error" => "You must log in to access this page."}
+               }}}
+  end
+
+  test "renders breadcrumbs item", %{
+    account: account,
+    identity: identity,
+    conn: conn
+  } do
+    {:ok, _lv, html} =
+      conn
+      |> authorize_conn(identity)
+      |> live(~p"/#{account}/settings/dns")
+
+    assert item = Floki.find(html, "[aria-label='Breadcrumb']")
+    breadcrumbs = String.trim(Floki.text(item))
+    assert breadcrumbs =~ "DNS Settings"
+  end
+
+  test "renders form", %{
+    account: account,
+    identity: identity,
+    conn: conn
+  } do
+    {:ok, lv, _html} =
+      conn
+      |> authorize_conn(identity)
+      |> live(~p"/#{account}/settings/dns")
+
+    form = lv |> form("form")
+
+    assert find_inputs(form) == [
+             "configuration[clients_upstream_dns][0][_persistent_id]",
+             "configuration[clients_upstream_dns][0][address]"
+           ]
+  end
+
+  test "saves custom DNS server address", %{
```

Comment:
You need a test the "can not be blank" error is not rendered when one of the list items was removed (test filter_errrors/3 behavior).

Reply:
Good call, will get a test written up.